### PR TITLE
Bug 1790629: use st.Manifest.OriginalFilename if object doesn't have a name

### PR DESCRIPTION
**Pull request by vrutkovs**
This is helpful when one of the manifests doesn't have a name (like `kind: List` in [image-registry PR](https://github.com/openshift/cluster-image-registry-operator/commit/d197841e7944138ee2dc0815f2749c4f67657889#diff-88bbb853bcd5a5fbc188c46767630fadR1))

This list still cannot be applied (`apply` requires a name to find existing resource) but it would be helpful to debug these kind of issues.

TODO:
* [x] Add unittest

### CODE REVIEW

**Comment by wking** [comment]
/approve

Ping me when there's a test, or you decide you don't need one ;).

**Comment by vrutkovs** [comment]
@wking added a few tests, PTAL

File: pkg/payload/task_test.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+						"kind":       "List",
+						"apiVersion": "v1",
+						"items": []interface{}{
+							"foo",
+							"bar",
+						},
+					},
+				},
+			}},
+			want: ` "list_manifest.yaml" (0 of 0)`,
```
Comment (wking) [inline pkg/payload/task_test.go]: Leading space when we have no GKV isn't great, but that's not new in this PR, and this is much better than the [current][1]:

```
Could not update list \"\" (188 of 529)"
```

for lists.

/lgtm

[1]: https://bugzilla.redhat.com/show_bug.cgi?id=1790629#c0

**Pull request merged**